src/tla/Typing.tla

```diff
@@ -46,4 +46,16 @@ AssumeType(name, tp) == TRUE
 (****************************************************************************)
 type_str :> ex == ex
 
+(****************************************************************************)
```

Comment:
fwiw, it seems to me it would be much nicer to just have an operator that let people make inline type annotations. Having to add adhoc value constructors as special cases seems a bit smelly to me.

Reply:
Yes and no. I think it is a good idea to limit type annotations only to the declarations: variable declarations, constant declarations, and operator declarations. You can always make a workaround in the form of `LET foo: "Int => Int" :> myexpr IN foo`. However, the user expectations of what can be and should be annotated are much lower with the current approach. It is also often counterintuitive how one should treat an annotation of an arbitrary expression in the presence of polymorphism. I guess, that is why languages like Scala also have very limited scoping for type information.

Reply:
Arbitrary annotations also raise the issue of contravariance, which I totally fail to understand in the TLA+ context.

Reply:
Ok. Thanks for explaining your motivation.